api/modules/projects/controller.js

```diff
@@ -1,144 +1,158 @@
-var Boom = require('boom');
-var Promise = require('bluebird'); // jshint ignore:line
-
-var errors = require('../../classes/errors');
-var BaseController = require('../../classes/base_controller');
-var Publisher = require('../../classes/publisher');
-
-var Files = require('../files/model');
-var PublishedFiles = require('../publishedFiles/model');
-
-var Model = require('./model');
-var controller = new BaseController(Model);
-
-var dateTracker = require('../../../lib/utils').dateTracker;
-
-controller.formatRequestData = function(req) {
-  var now = new Date();
-  var data = {
-    title: req.payload.title,
-    user_id: req.payload.user_id,
-    tags: req.payload.tags,
-    description: req.payload.description,
-    date_created: now,
-    date_updated: now,
-    readonly: req.payload.readonly,
-    client: req.payload.client
-  };
-
-  // If it is an update request
-  if (req.params.id) {
-    data.id = parseInt(req.params.id);
-    delete data.date_created;
-  }
-
-  return data;
-};
+"use strict";
 
-controller.formatResponseData = dateTracker.convertToISOStrings();
+const Boom = require(`boom`);
+const Promise = require(`bluebird`);
 
-controller.create = function(req, reply) {
-  return BaseController.prototype.create.call(this, req, reply, dateTracker.convertToISOStrings(true));
-};
+const Errors = require(`../../classes/errors`);
+const BaseController = require(`../../classes/base_controller`);
+const Publisher = require(`../../classes/publisher`);
 
-controller.update = function(req, reply) {
-  return BaseController.prototype.update.call(this, req, reply, dateTracker.convertToISOStrings(true));
-};
+const FilesModel = require(`../files/model`);
+const PublishedFilesModel = require(`../publishedFiles/model`);
 
-controller.publishProject = function(req, reply) {
-  var result = Promise.resolve().then(function() {
-    var record = req.pre.records.models[0];
+const ProjectsModel = require(`./model`);
 
-    return Publisher.publish(record)
-    .then(dateTracker.convertToISOStrings(true))
-    .then(function(publishedRecord) {
-      return req.generateResponse(publishedRecord).code(200);
-    });
-  })
-  .catch(errors.generateErrorResponse);
+const DateTracker = require(`../../../lib/utils`).DateTracker;
 
-  return reply(result);
-};
+class PublishedFilesCleanup {
+  constructor(project) {
+    this.project = project;
+  }
 
-controller.unpublishProject = function(req, reply) {
-  var result = Promise.resolve().then(function() {
-    var record = req.pre.records.models[0];
+  _fetchPublishedFiles(fileModel) {
```

Comment:
supposed to signify that these methods aren't really meant to be called outside this class. They are internal implementation methods.